samples/subsys/shell/shell_module/src/main.c

```diff
@@ -29,41 +29,46 @@ void timer_expired_handler(struct k_timer *timer)
 
 K_TIMER_DEFINE(log_timer, timer_expired_handler, NULL);
 
-static void cmd_log_test_start(const struct shell *shell, size_t argc,
-			       char **argv, u32_t period)
+static int cmd_log_test_start(const struct shell *shell, size_t argc,
+			      char **argv, u32_t period)
 {
 	if (!shell_cmd_precheck(shell, argc == 1, NULL, 0)) {
```

Comment:
Shouldn't we go one step further and replace the calls to shell_cmd_precheck on the handlers with just a specific error so the shell would take care of calling shell_cmd_precheck internally?